### PR TITLE
SMA: Add speedwire interface

### PR DESCRIPTION
**Pull request by t-mon**
nymea-plugins pull request checklist:

- [x] Make sure the pull request's title is of format "Plugin name: Add support for xyz" or "New plugin: Plugin name"

- [x] Did you test the changes on hardware, if not (e.g. absence of required hardware), please mention a person to confirm it has been tested.

- [x] Did you update the plugin's README.md accordingly?

- [x] Did you update translations (`cd builddir && make lupdate`)?

- [x] If you added a new plugin, should it be added to nymea-plugins-all or nymea-plugins-maker?

### CODE REVIEW

File: sma/meta.json [hunk 1]
```diff
@@ -8,6 +8,6 @@
         "network"
     ],
     "categories": [
-
+        "Energy"
```
Comment (mzanetti) [inline sma/meta.json:11]: needs to be lowercase "energy"

**Review by mzanetti — CHANGES_REQUESTED** [review]

File: sma/speedwireinverterrequest.cpp [hunk 2]
```diff
@@ -0,0 +1,76 @@
+/* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
+*
+* Copyright 2013 - 2021, nymea GmbH
```
Comment (mzanetti) [inline sma/speedwireinverterrequest.cpp:3]: 22

**Review by mzanetti — APPROVED** [review]

File: sma/speedwirediscovery.cpp [hunk 3]
```diff
@@ -0,0 +1,348 @@
...
+        qCWarning(dcSma()) << "SpeedwireDiscovery: Unhandled data received" << datagram.toHex();
+        return;
+    }
+}
+
+void SpeedwireDiscovery::sendDiscoveryRequest()
+{
+    if (m_multicastSocket->writeDatagram(Speedwire::discoveryDatagramMulticast(), m_multicastAddress, m_port) < 0) {
+        qCWarning(dcSma()) << "SpeedwireDiscovery: Failed to send discovery datagram to multicast address" << m_multicastAddress.toString();
+        return;
```
[the remark this answers is not in the document]
Comment (t-mon) [reply sma/speedwirediscovery.cpp:324]: There is a dedicated timeout for this type of discovery, in any case, if the thing discovery info timouts before this one is finished, the object does not exist any more.

File: sma/integrationpluginsma.cpp [hunk 4]
```diff
@@ -192,3 +431,24 @@ void IntegrationPluginSma::onPlantOverviewReceived(const QString &messageId, Sun
...
     }
 }
+
+void IntegrationPluginSma::setupRefreshTimer()
+{
+    // If already set up...
+    if (m_refreshTimer)
+        return;
+
+    m_refreshTimer = hardwareManager()->pluginTimerManager()->registerTimer(2);
```
[the remark this answers is not in the document]
Comment (t-mon) [reply sma/integrationpluginsma.cpp:453]: Agreed, but it keeps the timers in the resource as clean as possible.

Also, the refresh method itself returns immediatly if there is a refresh cycle still in progress, so I think we are good regarding overloading.

**Pull request merged**